cmd/up/up.go

```diff
@@ -924,51 +924,56 @@ func (up *upContext) shutdownHybridMode() {
 		return
 	}
 
-	pid := up.hybridCommand.Process.Pid
-	if existProcess(pid) {
-		if err := killProcess(pid); err != nil {
-			oktetoLog.Warning("failed to stop gracefully local process related to command executed in hybrid mode: %s", err.Error())
+	pList, err := ps.Processes()
+	if err != nil {
+		oktetoLog.Warning("error getting list of process %v", err)
+		return
+	}
+
+	terminateChildProcess(up.hybridCommand.Process.Pid, pList)
+
+	if err := terminateProcess(up.hybridCommand.Process.Pid); err != nil {
+		oktetoLog.Debugf("error terminating process %s: %v", up.hybridCommand.Process.Pid, err)
+	}
+}
+
+func terminateChildProcess(parent int, pList []ps.Process) {
+	// assure all the child process are terminated when command is exited
+	for _, pR := range pList {
+		// skip when process is not child from parent
+		if pR.PPid() != parent {
+			continue
 		}
+		// iterate over the children of the parent
+		terminateChildProcess(pR.Pid(), pList)
 
-		waitTimeout := 15 * time.Second
-		hasFinished := hasHybridCommandFinished(pid, waitTimeout)
-		if !hasFinished {
-			oktetoLog.Warning("timeout waiting to finish hybrid command gracefully")
+		if err := terminateProcess(pR.Pid()); err != nil {
+			if errors.Is(err, os.ErrProcessDone) {
+				continue
+			}
+			oktetoLog.Debugf("error terminating process %s: %v", pR.Pid(), err)
 		}
 	}
 }
 
-// killProcess sends a SIGTERM signal to the given process
-func killProcess(pid int) error {
+func terminateProcess(pid int) error {
 	p, err := os.FindProcess(pid)
 	if err != nil {
+		oktetoLog.Debugf("error getting process %s: %v", pid, err)
 		return err
 	}
-	return p.Signal(syscall.SIGTERM)
-}
-
-func existProcess(pid int) bool {
-	found, err := ps.FindProcess(pid)
-	if err == nil && found == nil {
-		return false
-	}
-	return true
-}
-
-func hasHybridCommandFinished(pid int, timeout time.Duration) bool {
-	ticker := time.NewTicker(1 * time.Second)
-	to := time.NewTicker(timeout)
-	for {
-		select {
-		case <-to.C:
-			return false
-		case <-ticker.C:
-			existProcess := existProcess(pid)
-			if !existProcess {
-				return true
-			}
+	if err := p.Signal(syscall.SIGTERM); err != nil {
+		if errors.Is(err, os.ErrProcessDone) {
```

Comment:
Do you think we should check the error here? I see it is handled [here](https://github.com/okteto/okteto/pull/3811/files#diff-c1b929a650f782cf212e9b4b0ced5a7a298598bcb7473be505e7af1f06d6cf30R951) so maybe we can send up the error and just checking it in the caller function. WDYT?